broker.go

```diff
@@ -248,9 +256,13 @@ func (b *Broker) muRefreshMetadata() error {
 
 // fetchMetadata is requesting metadata information from any node and return
 // protocol response if successful
+//
+// If "topics" are specified, only fetch metadata for those topics (can be
+// used to create a topic)
+//
 // Because it's using metadata information to find node connections it's not
 // thread safe and using it require locking.
-func (b *Broker) fetchMetadata() (*proto.MetadataResp, error) {
+func (b *Broker) fetchMetadata(topics []string) (*proto.MetadataResp, error) {
```

Comment:
What do you think about changing this function notation to accept `(topic ...string)`? I need to check if empty slice will work the same way as `nil`, but I think it's nicer to use with `...`